### PR TITLE
test: added integration tests

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                     |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✔                                                                       |

### CODE REVIEW

File: openapi3.yaml [hunk 1]
```diff
@@ -141,12 +141,18 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/error'
-  /mosaic:
+  /mosaic/{name}:
```
Comment (asafMasa) [inline openapi3.yaml:144]: if mosaic get is unimplemented - remove

File: openapi3.yaml [hunk 2]
```diff
@@ -165,6 +171,12 @@ paths:
       tags:
         - Mosaic
       summary: add layer to mosaic
+      parameters:
+        - name: name
+          in: path
+          required: true
+          schema:
+            type: string
```
Comment (asafMasa) [inline openapi3.yaml:179]: 201 response should be empty

File: openapi3.yaml [hunk 3]
```diff
@@ -208,6 +220,12 @@ paths:
       tags:
         - Mosaic
       summary: update mosaic layers
+      parameters:
+        - name: name
+          in: path
+          required: true
+          schema:
+            type: string
```
Comment (asafMasa) [inline openapi3.yaml:228]: remove response body from 204

File: openapi3.yaml [hunk 4]
```diff
@@ -208,6 +220,12 @@ paths:
       tags:
         - Mosaic
       summary: update mosaic layers
+      parameters:
```
Comment (asafMasa) [inline openapi3.yaml:223]: add to description (above all other layers)

File: tests/integration/layers/layersManager.spec.ts [hunk 5]
```diff
@@ -38,4 +44,196 @@ describe('layerManager', function () {
...
+  describe('#addLayer', function () {
+    it('Happy Path', async function () {
+      const response = await requestSender.addLayer(mockLayerNameIsNotExists);
+
+      expect(response.status).toBe(httpStatusCodes.CREATED);
+    });
+
+    it('Bad Path', async function () {
+      const mockBadRequestRequest = ({
+        mockName: 'NameIsNotExists',
```
Comment (asafMasa) [inline tests/integration/layers/layersManager.spec.ts:57]: add comment of invalid input parameter name

File: tests/integration/layers/layersManager.spec.ts [hunk 6]
```diff
@@ -38,4 +44,196 @@ describe('layerManager', function () {
...
+      const mockMosaicName = 'mosaicMockName';
+      const mockBadRequestRequest = ({
+        mockName: 'layerNameIsNotExists',
+      } as unknown) as ILayerToMosaicRequest;
+      const response = await requestSender.addLayerToMosaic(mockMosaicName, mockBadRequestRequest);
+
+      expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
+    });
+
+    it('Sad Path', async function () {
```
Comment (asafMasa) [inline tests/integration/layers/layersManager.spec.ts:150]: explain 
"Sad Path - mosaic not exist"
"Sad Path - layer not exist"
etc.

File: tests/integration/layers/layersManager.spec.ts [hunk 7]
```diff
@@ -38,4 +44,196 @@ describe('layerManager', function () {
...
+          { mockName: 'amsterdam_5cm', zIndex: 1 },
+          { mockName: 'LayerNameIsNotExists', zIndex: 0 },
+        ],
+      } as unknown) as IUpdateMosaicRequest;
+      const response = await requestSender.updateMosaic(mockMosaicName, mockBadRequest);
+
+      expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
+    });
+
+    it('Sad Path', async function () {
```
Comment (asafMasa) [inline tests/integration/layers/layersManager.spec.ts:206]: explain as above

**Review by asafMasa — CHANGES_REQUESTED** [review]
thanks

File: openapi3.yaml (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply openapi3.yaml:140]: fixed in 66bbf55

File: openapi3.yaml (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply openapi3.yaml:196]: fixed in 5ea5f76

File: openapi3.yaml (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply openapi3.yaml:151]: fixed in 5ea5f76

File: openapi3.yaml (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply openapi3.yaml:191]: fixed in 5ea5f76

File: tests/integration/layers/layersManager.spec.ts (hunk 7 above)
Comment (CL-SHLOMIKONCHA) [reply tests/integration/layers/layersManager.spec.ts]: fixed in 2928d1e

File: tests/integration/layers/layersManager.spec.ts (hunk 6 above)
Comment (CL-SHLOMIKONCHA) [reply tests/integration/layers/layersManager.spec.ts]: fixed in 2928d1e

File: tests/integration/layers/layersManager.spec.ts (hunk 5 above)
Comment (CL-SHLOMIKONCHA) [reply tests/integration/layers/layersManager.spec.ts:58]: fixed in 9b9d912

**Review by asafMasa — APPROVED** [review]

**Pull request merged**